src/components/Character/Detail.jsx

```diff
@@ -0,0 +1,34 @@
+import React, { useEffect, useState } from 'react';
+import { useParams, useHistory } from 'react-router-dom';
+import { fetchId } from '../../services/rickapi';
+
+export default function CharacterDetail() {
+  const { id } = useParams();
+  const [info, setInfo] = useState({});
+  const history = useHistory('');
+
+  useEffect(() => {
+    const fetchInfo = async () => {
+      const data = await fetchId(id);
+      setInfo(data);
+    };
+    fetchInfo();
+  }, [id]);
+
+  const goBack = async () => {
+    history.push('/');
+  };
+
+  return (
+    <article>
+      <img alt="Image of character" src={info.image} />
```

Comment:
It looks like in your app that when you navigate to your details that the fetch may be lagging. The `img` is broken initially then renders, could totally be an API thing as well.